lib/providers/products_provider.dart

```diff
@@ -94,9 +95,19 @@ class ProductsProvider with ChangeNotifier {
     }
   }
 
-  ///Deletes a product in the list
-  void deleteProduct(String id) {
-    _items.removeWhere((prod) => prod.id == id);
+  ///Deletes a product in the list and in the database
+  Future<void> deleteProduct(String id) async {
+    final url = 'https://shop-app-4ed38-default-rtdb.firebaseio.com/products/$id.json';
```

Comment:
you could use your `endpoint` from constants here, then just append `/products/$id.json`